CHANGELOG.md

```diff
@@ -16,6 +16,12 @@ should change the heading of the (upcoming) version to include a major version b
 
 -->
 
+# 5.6.4
```

Comment:
This looks like it might actually be a breaking change unless there is a way to figure out whether the user is using fluent 7 vs 8 and make adjust the props accordingly.

If it really is a breaking change, I'm tempted to hold of doing this until we begin working on v6 since we were planning on upgrading `antd`, `chakra-ui`, `bootstrap-4` and `semantic-ui` to the next major versions in that release as well. How quickly do you need this @saip92 ??

Reply:
@heath-freenome I've did some testing and made a couple of changes to ensure the theme still works with `@fluentui/react: >= 7`.

The changes are mostly to get typescript to be happy since a change in props would really just be no-op depending on the version.

The only specific workaround is for the `UpDownWidget` where I am checking the event passed for the `OnChange` callback, which I tested locally for both v7 and v8, and can confirm works as expected.

Reply:
@saip92 Awesome work! Can you also update the playground to upgrade fluent-ui there too? And, if you don't mind, maybe figure out why the styling hasn't quite been working lately? If that last request is too much, don't worry about it since it has been a bit busted for a while now.

Reply:
Sure. Will do. I did take a look at the playground but wasn't really sure what was the problem.

I was thinking of getting this PR merged first and then taking a look into that again later.

Reply:
@heath-freenome The problem is discussed [here](https://github.com/microsoft/fluentui/issues/13502) but unfortunately has no solution. I believe this will be fixed with v9 since [the way styles are applied](https://react.fluentui.dev/?path=/docs/concepts-migration-getting-started--page#build-time-styles) has changed.

So, I got a crazy idea and tried out using [the SSR approach](https://github.com/microsoft/fluentui/wiki/Server-side-rendering-and-browserless-testing#nodejs-setup) to inject the styles into the `Frame`, which did get the styling to work for most components, but they aren't interactive as a result (screenshots below). I suppose [the path to using Next.JS for the playground](https://github.com/rjsf-team/react-jsonschema-form/pull/3589) would eventually resolve this.

![image](https://github.com/rjsf-team/react-jsonschema-form/assets/897801/22e1bd50-f247-43c3-bb51-e60ff7446801)

Here is what the code looks like
![image](https://github.com/rjsf-team/react-jsonschema-form/assets/897801/1baf987f-ab95-4ed4-9f06-05aa744bad4b)

I realize lack of interactivity is a problem, especially for dropdowns, calendars, etc. and not all components work like the datetime field (not sure why) but is a workaround in the meantime to get a feel for the UI.

What do you think?

Reply:
@saip92 The lack of interactivity is a big issue. Let's get this PR merged and then hopefully you'll figure out a fully working fix. Can you rebase this PR onto main and resolve the conflict?

Reply:
@heath-freenome I understand. Rebased onto `main` and pushed the changes. Resolving this converstation.